graphite/graphite.go

```diff
@@ -0,0 +1,188 @@
+package graphite
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdb/influxdb"
+)
+
+var (
+	// ErrBindAddressRequired is returned when starting the Server
+	// without a TCP or UDP listening address.
+	ErrBindAddressRequired = errors.New("bind address required")
+
+	// ErrServerClosed return when closing an already closed graphite server.
+	ErrServerClosed = errors.New("server already closed")
+
+	// ErrDatabaseNotSpecified retuned when no database was specified in the config file
+	ErrDatabaseNotSpecified = errors.New("database was not specified in config")
+
+	// ErrServerNotSpecified returned when Server is not specified.
+	ErrServerNotSpecified = errors.New("server not present")
+)
+
+type ProcessorSink interface {
+	WriteSeries(database, retentionPolicy, name string, tags map[string]string, timestamp time.Time, values map[string]interface{}) error
+	DefaultRetentionPolicy(database string) (*influxdb.RetentionPolicy, error)
+}
```

Comment:
It probably makes more sense to allow `influxdb.Server.WriteSeries()` to allow a blank retention policy name and look it up inside the `WriteSeries()` function instead of requiring the caller to implement that.

In that case, we can remove the `DefaultRetentionPolicy()` from this interface and rename it:

```
type SeriesWriter interface {
	WriteSeries(database, retentionPolicy, name string, tags map[string]string, timestamp time.Time, values map[string]interface{}) error
}
```